### PR TITLE
Fixed additional edge cases with trailing comments and whitespace

### PR DESCRIPTION
**Pull request by afalko**

### CODE REVIEW

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 1]
```diff
@@ -18,6 +19,8 @@
 import java.nio.file.Paths;
 import java.util.List;
 
+import static java.lang.Character.SPACE_SEPARATOR;
```
Comment (brian-mcnamara) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java]: Is this used?

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 package com.salesforce.dockerfileimageupdate.utils;
 
+import com.google.common.base.CharMatcher;
```
Comment (brian-mcnamara) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java]: This too looks unused.

**Comment by afalko** [comment]
Thanks @brian-mcnamara ! I had my ide refactor imports and it found a few more.

File: dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtilTest.java [hunk 3]
```diff
@@ -404,25 +404,36 @@ public void testFindImagesAndFix_doNotDeleteOtherLines() throws Exception {
         assertEquals(strB.toString(), "hello\nFROM image:newtag\nthis is a test\n\n\n\nworld\n");
     }
 
-    @Test
-    public void testFindImagesAndFix_doNotDeletePostTagData() throws Exception {
+    @DataProvider
+    public Object[][] postTagData() {
```
Comment (justinharringa) [inline dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtilTest.java]: Wouldn't it be better to do this kind of testing directly against `getEndOfTagLength(String)` to keep the locus of test smaller and narrow scope (smaller unit for the unit test)?

**Comment by afalko** [comment]
@justinharringa I found additional edge cases that we were missing and added a separate test for the length finder. I fixed three bugs with this exercise: 
1. Tag would get wiped out if tab followed comment.
2. You can now have a comment with a tag: `image:tag # :commented-out-tag`
3. Empty tag returns the right thing
